### ISSUE TITLE
ocamlformat version not available in opam and not compatible with recent OCaml versions

### ISSUE THREAD

**Issue by cdisselkoen** [facebook/infer#1473]
Infer version: `v1.1.0-e3685a8d9`
OS: Ubuntu 18.04, no Docker
Command: `make BUILD_MODE=dev-noerror -j8 config_tests`
If it matters, I originally installed Infer with `./build-infer.sh clang`; that is, no Java support.

The command above fails with over 4000 lines of output, most of which is many copies of the error
```
ocamlformat: Error while parsing .ocamlformat:
             For option "version": expecting "0.18.0" but got "0.14.1-13-g000da99"
             ocamlformat src/unit/.formatted/<file>.ml (exit 1)
```

I do indeed have `ocamlformat 0.18.0` installed on my Infer opam switch.  The same error happens with other `ocamlformat` versions.  I can't use `0.14.1` because I am on OCaml 4.12.0, and that is not compatible with `ocamlformat 0.14.1`.  Further, the version `0.14.1-13-g000da99` is not available from the main `opam` repository.

**Comment by jvillard** [comment]
Yes, setting up ocamlformat for infer is quite a pain at the moment, sorry about that. The real solution is for us to upgrade to a public version of ocamlformat. This is in the works.

In the meantime, the solution is quite involved, but doable: you can install ocamlformat 0.14.1-13-g000da99 by manually pinning the "nebuchadnezzar" branch from ocamlformat's GitHub on an older opam switch, e.g. 4.09.1. You should be able to copy that binary somewhere without repeating this step afterwards. Not sure how to do that sensibly in Docker though.

**Comment by jvillard** [comment]
Should be fixed with 859d69c8bd30e3fc02a9394b2da9da1fe5368993: running `make devsetup` will install ocamlformat at the expected version.
